app/models/ability.rb

```diff
@@ -0,0 +1,9 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    if user.present?
+      docs = Permission.all.where(user_id: user.id)
+      can :manage Document id: docs
```

Comment:
i think you can remove this line and use `can :manage, Document, user_id: user.id`